app/models/resource_description_step.rb

```diff
@@ -19,8 +19,13 @@ def initialize(step = 'resource-description', title = "Resource description", su
 
   def execute context
     mediaobject = context[:mediaobject]
-    mediaobject.permalink = context[:media_object].delete(:permalink)
-    mediaobject.update_datastream(:descMetadata, context[:media_object])
+    populate_from_catalog = context[:media_object][:import_bib_record].present?
+    if populate_from_catalog
+      mediaobject.descMetadata.populate_from_catalog!(context[:media_object][:bibliographic_id])
+    else
+      mediaobject.permalink = context[:media_object].delete(:permalink)
```

Comment:
I think I get it...it is included in the else because the step is getting called two times: import and save.  So the else clause should run on the save?